feature-libs/organization/administration/components/budget/form/budget-form.component.spec.ts

```diff
@@ -26,19 +29,22 @@ const mockForm = new FormGroup({
   budget: new FormControl(),
 });
 
+let activeUnitList$: BehaviorSubject<B2BUnitNode[]>;
+let currencies$: BehaviorSubject<Currency[]>;
+
 class MockOrgUnitService {
-  getActiveUnitList() {
-    return of([]);
-  }
+  getActiveUnitList = () => activeUnitList$.asObservable();
   loadList() {}
 }
 
 class MockCurrencyService {
-  getAll() {}
+  getAll = () => currencies$.asObservable();
 }
 
 class MockItemService {
-  getForm() {}
+  getForm() {
+    return mockForm;
```

Comment:
even you re-create component in beforeEach and it will get new instance of BudgetItemService it will always return the same mockForm object instance. So that's the reason that field 'uid' was already set